services/wireguard/endpoint/endpoint.go

```diff
@@ -68,6 +63,29 @@ func (ce *connectionEndpoint) Start(config *wg.ServiceConfig) error {
 		ce.privateKey = config.Consumer.PrivateKey
 	}
 
+	if ce.ipResolver != nil {
```

Comment:
This shows too much details - could be separate private function with clear name